File: frontend/src/types/Task.ts
export type Task = {
  id: string;
  details?: string;
  completed: boolean;
  createdDate?: string;
  updatedDate?: string;
} & CreateTask;

export type CreateTask = {
  name: string;
};

export type UpdateTask = {
  id:string;
}

export type DeleteTask = {
  id:string;
}


export type ToggleTaskCompletion = {
  id:string;
}

File: backend/src/main/java/com/example/backendapplication/model/Task.java
package com.example.backendapplication.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;
import java.util.ArrayList;
import java.util.UUID;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tasks")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;

    private String details;

    private boolean completed;

    private Date createdDate;

    private Date updatedDate;

    @OneToMany(mappedBy = "task", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<SubTask> subtasks = new ArrayList<>();
}
